libs/shared/router/src/lib/protected-route/protected-route.tsx

```diff
@@ -9,12 +9,14 @@ export interface IProtectedRoute {
 
 export const ProtectedRoute = ({ children }: IProtectedRoute) => {
   const { checkIsAuthenticated, getCurrentUser } = useAuth()
+  const location = useLocation()
 
   useEffect(() => {
     getCurrentUser()
   }, [getCurrentUser])
 
   if (!checkIsAuthenticated) {
+    localStorage.setItem('redirectLoginUri', location.pathname)
```

Comment:
Same asserting that we are well setting the item in the local storage when an user is not authenticated could be a nice non regression test.